tarea_1.py

```diff
@@ -57,10 +57,363 @@
 la tarea.
 
 """
-__author__ = 'escribe_tu_nombre'
 
-import entorno_o
+##############################################################
+
+__author__ = 'IvanAlejandroMorenoSoto'
+
+##############################################################
+
+import entornos_o
+from random import random, choice
+from doscuartos_o import DosCuartos, AgenteReactivoModeloDosCuartos, AgenteAleatorio
+
+##############################################################
+
+# Ejercicio 1.
+
+class SeisCuartos(entornos_o.Entorno):
+    """
+    Entorno de una casa con seis cuartos: tres en la planta inferior y
+    tres en la superior.
+
+    Análogamente a DosCuartos, el estado se define como:
+    estado := [posición, A, B, C, D, E, F]
+
+    D E F
+    A B C
+
+    Donde A, B, C, son los cuartos inferiores, D, E, F, los superiores,
+    y posición puede tomar como valor cualquiera de ellos. Cada cuarto
+    puede estar "limpio" o "sucio."
+
+    Las acciones válidas son:
+    acciones = {"ir_Derecha", "ir_Izquierda", "subir", "bajar", "limpiar", "nada"}
+    Todas son legales en todos los cuartos excepto por "subir" que únicamente es
+    legal en A y C, y "bajar" que sólo se puede realizar en E.
+
+    Los sensores son una tupla que contiene la posición del robot y el estado de
+    limpieza del cuarto.
+    """
+
+    def __init__(self, x0=["A", "sucio", "sucio", "sucio", "sucio", "sucio", "sucio"]):
+        """
+        Define el estado inicial de este entorno.
+        De forma predeterminada el robot se encuentra en el cuarto inferior izquierdo
+        y toda la casa está sucia.
+
+        @param x0: Vector con el estado inicial del entorno de la forma
+        [posiciónInicial, limpieza_A, limpieza_B, limpieza_C, limpieza_D, limpieza_E, limpieza_F].
+        """
+        self.x = x0[:]
+        self.desempeño = 0
+
+    def acción_legal(self, acción):
+        """
+        Determina si una acción es legal en el estado actual.
+
+        @param acción: Acción que será revisada.
+
+        @return True si la acción es legal, False en caso contrario.
+        """
+        # Se separan los casos en: el robot quiere subir o quiere bajar o quiere hacer
+        # cualquier otra cosa.
+        if acción == "subir" and (self.x[0] == "A" or self.x[0] == "C"):
```

Comment:
Es más pythonudo (pero está perfectamente bien)

if acción == 'subir and self.x[0] in ["A", "B"]: